types.d.ts

```diff
@@ -0,0 +1,241 @@
+/**
+ * The nil UUID string (all zeros)
+ *
+ * @example
+ * ```javascript
+ * import { NIL as NIL_UUID } from 'uuid';
+ *
+ * NIL_UUID; // ⇨ '00000000-0000-0000-0000-000000000000'
+ * ```
+ */
+export const NIL: '00000000-0000-0000-0000-000000000000'
+
+/**
+ * Convert UUID string to array of bytes
+ *
+ * @param {string} str - A valid UUID string
+ * @throws {TypeError} If `str` is not a valid UUID
+ *
+ * @note Ordering of values in the byte arrays used by `parse()` and `stringify()` follows the left ↠ right order of hex-pairs in UUID strings. As shown in the example below.
+ *
+ * @example
+ * ```javascript
+ * import { parse as uuidParse } from 'uuid';
+ *
+ * // Parse a UUID
+ * const bytes = uuidParse('6ec0bd7f-11c0-43da-975e-2a8ad9ebae0b');
+ *
+ * // Convert to hex strings to show byte order (for documentation purposes)
+ * [...bytes].map((v) => v.toString(16).padStart(2, '0')); // ⇨
+ *   // [
+ *   //   '6e', 'c0', 'bd', '7f',
+ *   //   '11', 'c0', '43', 'da',
+ *   //   '97', '5e', '2a', '8a',
+ *   //   'd9', 'eb', 'ae', '0b'
+ *   // ]
+ * ```
+ */
+export function parse(str: string): Uint8Array
```

Comment:
What's the reasoning behind exposing the Uint8Array type like this? Could we instead return a value of type `Uuid` instead, to avoid leaking the fact that this is just bytes?

Reply:
The `parse()` method is conceptually simple: Parse a UUID string into it's constituent bytes. This isn't going to change anytime soon.  `Uint8Array` is well understood to be the proper JS type for such things and is, in fact, what the method returns.